internal/controllers/topology/machinedeployment/machinedeployment_controller.go

```diff
@@ -55,14 +56,23 @@ type Reconciler struct {
 }
 
 func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
-	err := ctrl.NewControllerManagedBy(mgr).
+	clusterToMachineDeployments, err := util.ClusterToTypedObjectsMapper(mgr.GetClient(), &clusterv1.MachineDeploymentList{}, mgr.GetScheme())
+	if err != nil {
+		return err
+	}
+
+	err = ctrl.NewControllerManagedBy(mgr).
 		For(&clusterv1.MachineDeployment{}).
 		Named("topology/machinedeployment").
-		WithOptions(options).
 		WithEventFilter(predicates.All(ctrl.LoggerFrom(ctx),
 			predicates.ResourceNotPausedAndHasFilterLabel(ctrl.LoggerFrom(ctx), r.WatchFilterValue),
 			predicates.ResourceIsTopologyOwned(ctrl.LoggerFrom(ctx)),
```

Comment:
I think this predicate has to be only for MD's. The Cluster itself is not topology owned. But it's enough to only move this predicate into the For

Reply:
I was somehow under the impression ResourceNotPaused checks both `spec.paused` and annotations.


```
	err = ctrl.NewControllerManagedBy(mgr).
		For(&clusterv1.MachineDeployment{},
			builder.WithPredicates(
				predicates.ResourceNotPaused(ctrl.LoggerFrom(ctx))),
		).
		Owns(&clusterv1.MachineSet{}).
		Watches(
			&clusterv1.MachineSet{},
			handler.EnqueueRequestsFromMapFunc(r.MachineSetToDeployments),
		).
		WithOptions(options).
		WithEventFilter(predicates.ResourceHasFilterLabel(ctrl.LoggerFrom(ctx), r.WatchFilterValue)).
		Watches(
			&clusterv1.Cluster{},
			handler.EnqueueRequestsFromMapFunc(clusterToMachineDeployments),
			// TODO: should this wait for Cluster.Status.InfrastructureReady similar to Infra Machine resources?
			builder.WithPredicates(
				predicates.ClusterUnpaused(ctrl.LoggerFrom(ctx)),
			),
		).Complete(r)
```

Reply:
In `ResourceNotPausedAndHasFilterLabel` the `ResourceNotPaused`  wouldn't be applicable for Clusters though. I will also separate that.